internal/server/http/handler/v1/tree/handler.go

```diff
@@ -90,10 +107,36 @@ func GetTreeByID(svc service.TreeService) fiber.Handler {
 // @Security		Keycloak
 func GetTreeBySensorID(svc service.TreeService) fiber.Handler {
 	return func(c *fiber.Ctx) error {
-		return getTreeByField(c, "sensor_id", svc.GetBySensorID)
+		ctx := c.Context()
+		id := c.Params("sensor_id")
+		if !isValidSensorID(id) {
+			err := service.NewError(service.BadRequest, "invalid sensor ID")
+			return errorhandler.HandleError(err)
+		}
+		domainData, err := svc.GetBySensorID(ctx, id)
+		if err != nil {
+			return errorhandler.HandleError(err)
+		}
+		data := mapTreeToDto(domainData)
+
+		return c.JSON(data)
 	}
 }
 
+func isValidSensorID(sensorID string) bool {
+	specialChars := `@#$%^&*`
+	containsSpChars := strings.ContainsAny(sensorID, specialChars)
+	invalidLength := len(sensorID) < 3 || len(sensorID) > 36
+	hasWhiteSPaces := strings.ContainsAny(sensorID, " \t\n")
+	htmlPattern := regexp.MustCompile(`(?i)<[a-z]+[^>]*>.*</[a-z]+>|<.*>`)
+	isLikeHTML := htmlPattern.MatchString(sensorID)
+	if containsSpChars || invalidLength || hasWhiteSPaces || isLikeHTML {
+		return false
+	}
+
+	return true
+}
```

Comment:
I don't know if we need this level of detail about a valid sensor ID. I think a simple regex expression for our sensor id type is sufficient. We should also make this a global function inside the sensor entity if you really want to check if the sensor id is valid.

Also, why is this function only used here to validate the sensor ID, not in the sensor handler it self?

Reply:
What do think the validation should look like (simple regex)?
I believe that the validation is very important, especially for a string-id.